src/regdesk/communication_manager.js

```diff
@@ -42,26 +46,26 @@ export class CommunicationManager {
    * @param {string} searchString - The string to search for.
    * @return {Promise<RegistrantDetails[]>} The promise that will return the search results.
    */
-  startSearchForRegByName(searchString) {
-    // Start the search process, internally storing the promise and handling
-    // the return value. Something like:
-    // this.#searhPromise = this
-    //   .whateverCommunicationThing.actuallyDoTheSearch(searchString)
-    //   .then(handleSearchResults);
-
-    // We then expect the LoadingState to add its own .then to the promise and
-    // handle it appropriately.
-
-    // In the meantime, mock it out
-    const arr = [];
-    const len = Math.floor(Math.random() * (4));
-    for (let i = 0; i < len; i++) {
-      arr.push(this.#testGetRandomReg());
-    }
-
-    return new Promise((resolve) => setTimeout(resolve, 500))
-      .then(() => arr) // The mock one!
-      .then(this.handleSearchResults.bind(this)); // The real one!
+  async startSearchForRegByName(searchString) {
+    // FIXME I think we still need to do some client side filtering for cancelled/duplicate/etc memberships.
+    const bearerToken = await this.#loginMgr.getBearerToken();
+    return RegfoxApi.searchRegistrations(searchString, bearerToken)
+      .then((result) => result.registrants.slice(0, 10)) // The next call hammers regfox, so we want to limit this as much as possible
+      .then((results) => Promise.all(results.map((result) => RegfoxApi.getRegistrationInfo(result.id, bearerToken))))
+      .then((results) => results.map((result) => new RegistrantDetails({
+        preferredName: result.preferredFirstName || result.name,
+        legalName: result.name,
+        birthdate: result.dateOfBirth,
+        amountDue: result.outstandingAmount100x, // FIXME there is a bug where #maybeSingleResultToUnpaid
+        // is unable to view this value, and therefor thinks everyone is unpaid.
+        badgeLine1: result.badgeLine1Text,
+        badgeLine2: result.badgeLine2Text || '',
+        badgeId: '12345678', // I have no idea which of id is this one.
```

Comment:
The badgeID was the 8 digit value that was pulled out of the browser's URL, I'm not sure which actual API value this maps to.

Reply:
Do you have an example URL, like from the test account? I can figure it out from there. :)

Reply:
Found one!

https://manage.webconnex.com/a/1311/reports/orders/23028022/details/26564608

The 26564608 is the BadgeId, and which is the same as `registrant.id`, will update!